sips/cluster_consensus.md

```diff
@@ -0,0 +1,440 @@
+|     Author     |           Title            |  Category  |       Status        |    Date    |
+| -------------- | -------------------------- | ---------- | ------------------- | ---------- |
+| Matheus Franco, Gal Rogozinski | Committee consensus          | Core       | open-for-discussion | 2024-03-05 |
+
+## Summary
+
+Aggregate `Attestation` and `Sync Committee` duties based on the committee of operators and the duties' slot. 
+
+## Motivation
+
+With the current design, a committee of operators associated with several validators may end up performing more than one attestation or sync committee duties on equivalent data. This proposal helps to decrease the number of messages exchanged in the network and the processing cost.
+
+## Rationale
+
+The aggregation of duties is possible because the data that must be agreed on is independent of the validator.
+
+For example, take a look at the `AttestationData` type.
+```go
+type AttestationData struct {
+	Slot            Slot
+	Index           CommitteeIndex
+	BeaconBlockRoot Root `ssz-size:"32"`
+	Source          *Checkpoint
+	Target          *Checkpoint
+}
+```
+The only validator-dependent field is `CommitteeIndex` and it does not have to be agreed on.
+
+For the `Sync Committee` duty, operators agree on a `phase0.Root` data which is also independent of the validator.
+
+## Improvement
+
+According to Monte Carlo simulations using a dataset based on the Mainnet, this proposal reduces to $21.60$% the current number of messages exchanged in the network. Note that this result includes aggregating the post-consensus messages into a single message.
+
+Regarding the number of bits exchanged, we estimate that this proposal will reduce the current value to, at least, $52.96$%. Notice that this reduction is not as significant as the number of messages reduction due to the larger post-consensus messages.
+
+Again with Monte Carlo simulations using the Mainnet dataset, the number of attestation duties aggregated presented the following distribution.
+
+<p align="center">
+<img src="./images/cluster_consensus/aggregated_duties.png"  width="50%" height="10%">
+</p>
+
+## Spec changes
+
+### Design
+
+Under the new design we will have a `Committee` object that will be the top level object in charge of processing consensus messages and partial signature messages for the attestation and sync committee roles.
+
+The `Committee` will hold a `CommitteeRunner` object for each slot it is running a beacon duty for.
+
+For other duty roles the old design will remain.
+
+#### Code
+
+```go
+// CommitteeDuty implements Duty
+type CommitteeDuty struct {
+	Slot         spec.Slot
+	BeaconDuties []*BeaconDuty
+}
+
+// Committee is a committee of a unique set of operators that have shared validators
+type Committee interface {
+	// Initializes and starts the runner for the duties for the given slot
+  	StartDuty(duty Duty) error
```

Comment:
@GalRogozinski should be CommitteeDuty, correct?

Reply:
`Duty` is an interface
But perhaps in the SIP it is indeed best to use `CommitteeDuty` for clarity

I will change